controller/src/constants.rs

```
@@ -15,3 +17,20 @@ pub(crate) fn requeue_slow() -> Action {
 pub(crate) fn no_requeue() -> Action {
     Action::await_change()
 }
+
+/// Parse a duration string into a Duration object.
+pub(crate) fn parse_duration(input: &str) -> Result<Duration, ParseIntError> {
```

This should return an error type from this crate.